File: app/src/main/java/com/adrzdv/mtocp/domain/repository/DepotRepository.java
package com.adrzdv.mtocp.domain.repository;

import com.adrzdv.mtocp.data.db.pojo.DepotWithBranch;

import java.util.List;

public interface DepotRepository {

    List<DepotWithBranch> getAll();

    DepotWithBranch getDepotById(int id);

    void saveAll(List<DepotWithBranch> list);

    List<DepotWithBranch> getDinnerDepots();
}

File: app/src/main/java/com/adrzdv/mtocp/domain/repository/DepotRepository.kt
package com.adrzdv.mtocp.domain.repository

import com.adrzdv.mtocp.data.db.pojo.DepotWithBranch

interface DepotRepository : BaseAppRepository<DepotWithBranch, Int> {
    suspend fun getDinnerDepots(): List<DepotWithBranch>
}